### PR TITLE
Added tx_power in advertisement_data for Windows, Mac and Linux

### PR DESCRIPTION
**Pull request by facepolish**
Pull Request Guidelines for Bleak
---------------------------------

Before you submit a pull request, check that it meets these guidelines:

1. If the pull request adds functionality, the docs should be updated.
2. Modify the `CHANGELOG.rst`, describing your changes as is specified by the
   guidelines in that document.
3. The pull request should work for Python 3.7+ on the following platforms:
    - Windows 10, version 16299 (Fall Creators Update) and greater
    - Linux distributions with BlueZ >= 5.43
    - OS X / macOS >= 10.11
4. Squash all your commits on your PR branch, if the commits are not solving
   different problems and you are committing them in the same PR. In that case,
   consider making several PRs instead.
5. Feel free to add your name as a contributor to the `AUTHORS.rst` file!

### CODE REVIEW

File: bleak/backends/bluezdbus/scanner.py [hunk 1]
```diff
@@ -231,13 +231,17 @@ def _handle_advertising_data(self, path: str, props: Device1) -> None:
         _service_data = {k: bytes(v) for k, v in props.get("ServiceData", {}).items()}
         _service_uuids = props.get("UUIDs", [])
 
+        # Get tx power data
+        tx_power = props.get("TxPower",None)
```
Comment (dlech) [inline bleak/backends/bluezdbus/scanner.py:235]: The default of `get()` is already `None`.
```suggestion
        tx_power = props.get("TxPower")
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -19,6 +19,8 @@ Changed
 * The BlueZ D-Bus backend implements a services cache between connections to significancy improve reconnect performance.
   To use the cache, call ``connect`` and ``get_services`` with the ``dangerous_use_bleak_cache``
   argument to avoid services being resolved again.
+* ``AdvertisementData`` class now has an attribute ``tx_power``. scanner.py in backends of winrt, corebluetooth and 
+  bluezdbus were changed to set tx_power if available.
```
Comment (dlech) [inline CHANGELOG.rst:23]: Can we get this for the Android backend too (even if you aren't able to test it)?

i.e. use `result.getTxPower()` in `_PythonScanCallback.onScanResult()`.

It also looks like we might want to convert `ScanResult.TX_POWER_NOT_PRESENT` to `None` to be consistent with other backends.

https://developer.android.com/reference/android/bluetooth/le/ScanResult#getTxPower()

File: bleak/backends/corebluetooth/scanner.py [hunk 3]
```diff
@@ -110,12 +110,16 @@ def callback(p: CBPeripheral, a: Dict[str, Any], r: int) -> None:
                 cb_uuid_to_str(u) for u in a.get("kCBAdvDataServiceUUIDs", [])
             ]
 
+            # set tx_power data if available
+            tx_power = a.get("kCBAdvDataTxPowerLevel",None)
```
Comment (dlech) [inline bleak/backends/corebluetooth/scanner.py:114]:
```suggestion
            tx_power = a.get("kCBAdvDataTxPowerLevel")
```

File: bleak/backends/scanner.py [hunk 4]
```diff
@@ -44,6 +45,9 @@ def __init__(self, **kwargs):
         # Tuple of platform specific data
         self.platform_data: Tuple = kwargs.get("platform_data", ())
 
+        # Tx Power data
+        self.tx_power: Optional[int] = kwargs.get("tx_power",None) # tx_power can't be 0. so 0 may be better default value?
```
Comment (dlech) [inline bleak/backends/scanner.py:49]:
```suggestion
        self.tx_power: Optional[int] = kwargs.get("tx_power")
```

The value can be from -127 to 126, so 0 is allowed. I think `None` makes sense for "not available".

**Review by dlech — CHANGES_REQUESTED** [review]
Thanks! I have made a few suggestions.

File: CHANGELOG.rst (hunk 2 above)
Comment (facepolish) [reply CHANGELOG.rst:23]: Yes. Actually now I have Android Studio but don't have test environment Android (will come in 2 weeks). So this time I made changes and test them later (if necessary). Thank you for your advice.

File: bleak/backends/p4android/scanner.py [hunk 5]
```diff
@@ -276,13 +276,18 @@ def onScanResult(self, result):
             entry.getKey().toString(): bytes(entry.getValue())
             for entry in record.getServiceData().entrySet()
         }
+        if result.TX_POWER_NOT_PRESENT:
+            tx_power = None
+        else:
+            tx_power = result.getTxPower() # get Tx Power
```
Comment (dlech) [inline bleak/backends/p4android/scanner.py:282]: Based on the android docs, I think this need to be:
```suggestion
        tx_power = result.getTxPower()

        # change "not present" value to None to match other backends
        if tx_power == defs.ScanResult.TX_POWER_NOT_PRESENT:
            tx_power = None
```

And we will need to add `ScanResult` to the `defs` module.

```python
ScanResult = autoclass("android.bluetooth.le.ScanResult")
```

Comment (dlech) [reply bleak/backends/p4android/scanner.py]: We still need to add `ScanResult = autoclass("android.bluetooth.le.ScanResult")` to `defs.py`

Comment (facepolish) [reply bleak/backends/p4android/scanner.py]: Thank you! Now I understand the structure of your code and usage of TX_POWER_NOT_PRESENT

**Review by dlech — APPROVED** [review]

**Comment by dlech** [comment]
Merged, thanks!

**Pull request merged**